### PR TITLE
feat(ci): Add tests into CI

### PR DESCRIPTION
**Pull request by Pothulapati**
part of #8 

This PR adds `make test` into the CI build pipeline. This
does not add any tests recently

### CODE REVIEW

File: internal/controller/dragonfly_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+			}, &svc)
+			Expect(err).To(BeNil())
+
+			// Wait until Dragonfly object is marked initialized
+			waitForDFInitialised(ctx, k8sClient, name, namespace, 2*time.Minute)
+			waitForStatefulSetReady(ctx, k8sClient, name, namespace, 2*time.Minute)
+
+			// Check if there are relevant pods with expected roles
+			var pods corev1.PodList
+			err = k8sClient.List(ctx, &pods, client.InNamespace("default"), client.MatchingLabels{
```
Comment (Pothulapati) [inline internal/controller/dragonfly_controller_test.go]: We are only testing for resource creation here, but we should also verify roles which I'm thinking can be in a separate PR

File: internal/controller/dragonfly_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+				resources.KubernetesPartOfLabelKey: "dragonfly",
+			})
+			Expect(err).To(BeNil())
+
+			Expect(pods.Items).To(HaveLen(4))
+		})
+	})
+})
+
+func waitForDFInitialised(ctx context.Context, c client.Client, name, namespace string, maxDuration time.Duration) error {
```
Comment (ashotland) [inline internal/controller/dragonfly_controller_test.go]: It's Dragonfly so Df or Dfly
But Dragonfly is best :)
